packages/devtools_app/lib/src/flutter/common_widgets.dart

```diff
@@ -336,6 +338,20 @@ Widget processingInfo({
   );
 }
 
+Widget exitOfflineButton(FutureOr<void> Function() onPressed) {
+  return OutlineButton(
+    key: const Key('exit offline button'),
+    onPressed: () {
+      offlineMode = false;
```

Comment:
why  is this state change outside a call to setState?

Reply:
removed and added a comment that the consumer will be responsible for this.